atak-civ/plugin-examples/plugintemplate/app/src/main/java/com/atakmap/android/plugintemplate/PresetTimerListAdapter.java

```diff
@@ -0,0 +1,183 @@
+        return  new PresetTimerListAdapter.ViewHolder(view);
+    }
+
+    /** From Android API Website (https://developer.android.com/guide/topics/ui/layout/recyclerview)
+     * RecyclerView calls this method to associate a ViewHolder with data.
+     * The method fetches the appropriate data and uses the data to fill in the view holder's layout.
+     * @param viewHolder the viewholder we created by inflating the timer_cell that we want to fill with data
+     * @param position  The position of the item within the recyclerview
+     */
+    @Override
+    public void onBindViewHolder(@NonNull final PresetTimerListAdapter.ViewHolder viewHolder, int position) {
+        // get corresponding timer
+        Timer currentTimer = preset_timers.get(position);
+
+        viewHolder.presetName.setText(currentTimer.getName());
+        viewHolder.presetDuration.setText(currentTimer.getDuration());
+
+        /*
+        sets the set button so that when it is clicked, the plugin returns to the home screen and
+        creates a new activeTimer corresponding to the current timer. Uses a deepcopy of the timer
+        so that changes to the preset timer don't affect the running ActiveTimer
+         */
+        viewHolder.setPresetButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // create clone of preset timer
+                Timer currentTimer = preset_timers.get(viewHolder.getAdapterPosition());
+                Timer timer = new Timer();
+                timer.setName(currentTimer.getName());
+                timer.setSound(currentTimer.getSound());
+                timer.setPreset(currentTimer.isPreset());
+                timer.setHours(currentTimer.getHours());
+                timer.setMinutes(currentTimer.getMinutes());
+                timer.setSeconds(currentTimer.getSeconds());
+                ArrayList<String> notificationsCopy = (ArrayList<String>) currentTimer.getNotifications().clone();
+                timer.setNotifications(notificationsCopy);
+
+                Intent i = new Intent();
+                i.setAction(PluginTemplateDropDownReceiver.SHOW_PLUGIN);
+                i.putExtra("TIMER", timer);
+                AtakBroadcast.getInstance().sendBroadcast(i);
+            }
+        });
+
+        /*
+        set the delete button so that when clicked it deletes the selected timer from the preset
+        view as well as the JSON file for preset timers
+        */
+        viewHolder.deletePresetButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Timer currentTimer = preset_timers.get(viewHolder.getAdapterPosition());
+                //TODO: delete timer from JSON file as well
+                int pos = preset_timers.indexOf(currentTimer);
+                preset_timers.remove(currentTimer);
+                notifyItemRemoved(pos);
```

Comment:
remove animation is quite weird, the timer above the one that is removed kind of "bounces" not sure if there's a way to fix

Reply:
I agree that the animation is weird. I'm not sure how to fix it right now but I think it's something Noah can probably look into when he works on the UI